app/screens/home/channel_list/servers/index.tsx

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Q} from '@nozbe/watermelondb';
+import React, {useCallback, useEffect, useRef, useState} from 'react';
+import {useIntl} from 'react-intl';
+import {StyleSheet} from 'react-native';
+
+import ServerIcon from '@components/server_icon';
+import {MM_TABLES} from '@constants/database';
+import {useServerUrl} from '@context/server';
+import {useTheme} from '@context/theme';
+import DatabaseManager from '@database/manager';
+import {useIsTablet} from '@hooks/device';
+import {bottomSheet} from '@screens/navigation';
+
+import ServerList from './servers_list';
+
+import type ServersModel from '@typings/database/models/app/servers';
+import type MyChannelModel from '@typings/database/models/servers/my_channel';
+import type {Subscription} from 'rxjs';
+
+type UnreadMessages = {
+    mentions: number;
+    unread: boolean;
+};
+
+type UnreadSubscription = UnreadMessages & {
+    subscription?: Subscription;
+}
+
+const {SERVERS} = MM_TABLES.APP;
+const {CHANNEL, MY_CHANNEL} = MM_TABLES.SERVER;
+const subscriptions: Map<string, UnreadSubscription> = new Map();
+
+const styles = StyleSheet.create({
+    icon: {
+        alignItems: 'center',
+        justifyContent: 'center',
+        position: 'absolute',
+        zIndex: 10,
+        top: 10,
+        left: 16,
+        width: 40,
+        height: 40,
+    },
+});
+
+export default function Servers() {
+    const db = DatabaseManager.appDatabase?.database;
+    const intl = useIntl();
+    const [total, setTotal] = useState<UnreadMessages>({mentions: 0, unread: false});
+    const registeredServers = useRef<ServersModel[]|undefined>();
+    const currentServerUrl = useServerUrl();
+    const isTablet = useIsTablet();
+    const theme = useTheme();
+
+    const updateTotal = () => {
+        let unread = false;
+        let mentions = 0;
+        subscriptions.forEach((value) => {
+            unread = unread || value.unread;
+            mentions += value.mentions;
+        });
+        setTotal({mentions, unread});
+    };
+
+    const unreadsSubscription = (serverUrl: string, myChannels: MyChannelModel[]) => {
+        const unreads = subscriptions.get(serverUrl);
+        if (unreads) {
+            let mentions = 0;
+            let unread = false;
+            myChannels.forEach((myChannel) => {
+                mentions += myChannel.mentionsCount;
+                unread = unread || myChannel.isUnread;
+            });
+
+            unreads.mentions = mentions;
+            unreads.unread = unread;
+            subscriptions.set(serverUrl, unreads);
+            updateTotal();
+        }
+    };
+
+    const serversObserver = async (servers: ServersModel[]) => {
+        registeredServers.current = servers;
+        servers.forEach((server) => {
```

Comment:
Should we also check for stale data in the `subscriptions` object? It should not be a "big" deal, since we won't receive events in the subscription from a deleted server, and the subscription will be unsubscribed eventually when the component gets dismounted. Either way, I though I should raise the question.

Reply:
can you elaborate a bit further?

Reply:
We have a `subscriptions` map at file level. If a server is removed, it is not removed from this component, so the only knowledge we have is that the `servers` object no longer has a server with a `url` present in `subscriptions`. I feel we are not handling the case were we remove the missing server from the `subscriptions` map.

Actually, now that I think about it, it is indeed a problem, since we may have stale unread and mention data in the `subscriptions` map, not only the subscription itself.

Reply:
interesting.. I'll give it a look, this will require changes in other components as well unrelated to this PR but I'll include it anyway